src/main/java/com/ernie/TicketApp/Ticket.java

```diff
@@ -0,0 +1,134 @@
+package com.ernie.TicketApp;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+import java.util.UUID;
+
+public class Ticket extends Entity {
+
+    @NullableWarning
+    private UUID id = UUID.randomUUID();
+
+    private String concertHall;
+    private int eventCode;
+    private LocalDateTime time;
+    private boolean isPromo;
+    private char stadiumSector;
+    private double maxBackpackWeightInKg;
+    private double price;
+    private LocalDateTime creationDateTime;
+
+    public Ticket(String concertHall, int eventCode, LocalDateTime time, boolean isPromo, char stadiumSector, double maxBackpackWeightInKg) {
+
+        if (concertHall.length() <= 10) {
+            this.concertHall = concertHall;
+        }
+        if (eventCode / 1000 < 1) {
+            this.eventCode = eventCode;
+        }
+        this.time = time;
+        this.isPromo = isPromo;
+        if (stadiumSector == 'A' || stadiumSector == 'B' || stadiumSector == 'C') {
+            this.stadiumSector = stadiumSector;
+        }
+        this.maxBackpackWeightInKg = maxBackpackWeightInKg;
+        if (isPromo) {
+            price = 10;
+        } else {
+            price = 20;
+        }
+        creationDateTime = LocalDateTime.now();
+    }
+
+    public Ticket() {
+        creationDateTime = LocalDateTime.now();
+    }
+
+    public Ticket(String concertHall, int eventCode, LocalDateTime time) {
+        if (concertHall.length() <= 10) {
+            this.concertHall = concertHall;
+        }
+        if (eventCode / 1000 < 1) {
+            this.eventCode = eventCode;
+        }
+        this.time = time;
+        price = 20;
+        creationDateTime = LocalDateTime.now();
+    }
+
+    public void shared(String phone) {
+        System.out.println("Ticket " + id + " was shared to phone number " + phone);
+    }
+
+    public void shared(String phone, String email) {
+        System.out.println("Ticket " + id + " was shared to phone number " + phone + " and to email " + email);
+    }
+
+    @Override
+    public String toString() {
```

Comment:
move it after equals & hashCode methods

Reply:
done in commit ba78624df638ea7b420b962d243bdefb20792f8d